flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/MySqlToDorisE2eITCase.java

```diff
@@ -336,6 +336,92 @@ public void testSyncWholeDatabase() throws Exception {
         }
     }
 
+    @Test
+    public void testSyncWholeDatabaseInBatchMode() throws Exception {
+        String databaseName = mysqlInventoryDatabase.getDatabaseName();
+        String pipelineJob =
+                String.format(
+                        "source:\n"
+                                + "  type: mysql\n"
+                                + "  hostname: mysql\n"
+                                + "  port: 3306\n"
+                                + "  username: %s\n"
+                                + "  password: %s\n"
+                                + "  tables: %s.\\.*\n"
+                                + "  server-id: 5400-5404\n"
+                                + "  server-time-zone: UTC\n"
```

Comment:
Do we need to add `scan.startup.mode: snapshot` in MySQL source config?

Reply:
It is currently needed. Since batch mode is only applicable to snapshots, should it be optimized so that snapshot is the default option if not specified?

Reply:
It will be great if we reduce this config, and we can extract pipeline config from Context when using `createDataSource(Context context)` method.